code/script.js

```diff
@@ -218,108 +227,240 @@ const generateBoard = () => {
           <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
         </div>
       </div>
-    `
-  })
+    `;
+  });
+};
+
+// Timer function
+function setTime() {
+  ++totalSeconds;
+  secondsLabel.innerHTML = pad(totalSeconds % 60);
+  minutesLabel.innerHTML = pad(parseInt(totalSeconds / 60));
 }
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
-const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+function pad(time) {
+  let timeString = time + "";
+  if (timeString.length < 2) {
+    return "0" + timeString;
+  } else {
+    return timeString;
+  }
 }
 
-// This function to start (and restart) the game
+// A randomly selected person from the characters array is set as the value of the variable called secret
+const setSecret = () => {
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
+
+// Function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  // All the characters will show from the beginning
+  charactersInPlay = CHARACTERS;
 
-// setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  // Restart timer
+  totalSeconds = 0;
+  valString = "";
+  secondsLabel.innerHTML = "";
+  minutesLabel.innerHTML = "";
+
+  // The board with the characters will show
+  generateBoard();
+
+  // One of the characters is randomly selected as the secret person
+  setSecret();
 
+  //Invokes the selectQuestion function
+  selectQuestion();
+
+  //The music restarts
+  gameMusic.play();
+  gameMusic.currentTime = 0;
+};
+
```

Comment:
Like all the comments, makes it easy to follow and remember what everything is and do :)